tests/test_models.py

```diff
@@ -27,6 +27,85 @@
 # TESTS
 # ------------------------------------------------------------------------
 
+def test_model_checkpoint_options():
```

Comment:
@Ir1d I updated the tests to reflect the desired behavior. Probably easier to modify the checkpoint accordingly until these tests pass.

Reply:
@Ir1d your fixes were off by one. the tests are correct. the implementation is off by 1 so needs to be fixed